crates/solidity/testing/solc/src/utils/binaries.rs

```diff
@@ -74,7 +81,19 @@ fn fetch_releases(mirror_url: &Url, binaries_dir: &Path) -> HashMap<Version, Str
     }
 
     let list_path = binaries_dir.join("list.json");
-    if !list_path.exists() {
+
```

Comment:
I think we don't have to `unwrap` so much here; this should probably be more resilient and self-documenting (and fixes a TOCTOU bug but it probably wouldn't ever bite us):

```diff
diff --git a/crates/solidity/testing/solc/src/utils/binaries.rs b/crates/solidity/testing/solc/src/utils/binaries.rs
index 52d562db8..1c15cc58f 100644
--- a/crates/solidity/testing/solc/src/utils/binaries.rs
+++ b/crates/solidity/testing/solc/src/utils/binaries.rs
@@ -7,8 +7,10 @@ use rayon::prelude::{ParallelBridge, ParallelIterator};
 use semver::Version;
 use serde::Deserialize;
 use std::{
-    collections::HashMap, os::unix::prelude::PermissionsExt, path::Path, path::PathBuf,
-    time::Duration,
+    collections::HashMap,
+    os::unix::prelude::PermissionsExt,
+    path::{Path, PathBuf},
+    time::{Duration, SystemTime},
 };
 use url::Url;
 
@@ -82,18 +84,10 @@ fn fetch_releases(mirror_url: &Url, binaries_dir: &Path) -> HashMap<Version, Str
 
     let list_path = binaries_dir.join("list.json");
 
-    let should_download_list = if list_path.exists() {
-        let created = list_path.metadata().unwrap().created().unwrap();
-        let one_day = Duration::from_secs(60 * 60 * 24);
+    let fetch_time = list_path.metadata().and_then(|metadata| metadata.created());
+    let older_than_day = |time| time + Duration::from_secs(60 * 60 * 24) < SystemTime::now();
 
-        // Download if if it is older than one day:
-        created.elapsed().unwrap() > one_day
-    } else {
-        // Doesn't exist. Download it:
-        true
-    };
-
-    if should_download_list {
+    if fetch_time.map(older_than_day).unwrap_or(true) {
         let list_url = mirror_url.join("list.json").unwrap();
         download_file(list_url, &list_path);
     }

```

Take from this what you will but I'd prefer if we keep some form of `list_path.metadata().and_then(|metadata| metadata.created())` to reduce unwraps/fix a race condition :pray:

Reply:
Sure.

Reply:
Fixed the bug, but I suggest using something like `anyhow` and `?` instead of callbacks, especially that we don't process the resulting errors.